### PR TITLE
RELATED: RAIL-3473 - Fixes & small tweaks discovered during KD integration

### PR DESCRIPTION
**Pull request by lupko**
-  rename bad event handling API method (removeEventCustomHandler -> removeCustomEventHandler)

-  dashboard event handling was susceptible to racy behavior; handlers would not be registered if the API was used after initialization of engine + plugins and before the dashboard component's event emitter got initialized

-  dashboard loader hook did not include unmount logic; plugins get their unload called during unmount

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5499/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7216/ : SUCCESS in 14m 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5500/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7217/ : SUCCESS in 15m 08s

File: libs/sdk-ui-loaders/src/dashboard/types.ts [hunk 1]
```diff
@@ -25,17 +25,30 @@ export interface IDashboardLoadOptions extends IDashboardBasePropsForLoader {
...
+     * The loader expects that it is running inside an application that depends on the `@gooddata/sdk-ui-dashboard`
+     * package. Furthermore the loader will initialize dashboard with only those plugins that are part of
+     * the application and passed via `extraPlugins` property. Plugins that are linked with the dashboard
+     * will be ignored.
      *
-     * The loading result is same regardless of the mode - it always includes everything that you need to
-     * render a dashboard in your application.
+     * This mode is suitable during plugin development
+     *
+     * `all` mode
```
Comment (no23reason) [inline libs/sdk-ui-loaders/src/dashboard/types.ts:37]: Outdated value, should be `adaptive`

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1823

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**